python/lsst/ts/epm/mib_tree.py

```diff
@@ -0,0 +1,185 @@
+    mib_tree[xups.name] = xups
+
+
+def _add_mib_elements() -> None:
+    global _line_num
+    for filename in DATA_DIR.glob("*.mib"):
+        with open(filename) as f:
+            lines = f.readlines()
+
+        for _line_num in range(len(lines)):
+            line = lines[_line_num].strip()
+            obj_id_match = re.match(OBJECT_IDENTIFIER, line)
+            obj_type_match = re.match(OBJECT_TYPE, line)
+            if obj_id_match:
+                parent_name = obj_id_match.group(2)
+                parent: MibTreeElement | None = _get_parent(parent_name)
+                assert parent is not None
+                name = obj_id_match.group(1)
+                if name == "synaccess":
+                    name = "pdu"
+                if name == "schneiderElectric":
+                    name = "scheiderPm5xxx"
+                mib_tree[name] = MibTreeElement(
+                    name=name,
+                    description=obj_id_match.group(1),
+                    oid=f"{parent.oid}.{obj_id_match.group(3)}",
+                    parent=parent,
+                    type=MibTreeElementType.BRANCH,
+                )
+            elif obj_type_match:
+                _process_obj_type(lines, line, obj_type_match)
+
+
+def _get_parent(parent_name: str) -> MibTreeElement | None:
+    parent: MibTreeElement | None = None
+    if parent_name == "xupsMIB":
+        parent_name = "xups"
+    elif parent_name == "synSys":
+        parent_name = "pdu"
+    elif parent_name == "schneiderElectric":
+        parent_name = "scheiderPm5xxx"
+    if parent_name in mib_tree:
+        parent = mib_tree[parent_name]
+    else:
+        log.debug(f"WOUTER {parent_name=!r} not found.")
+    return parent
+
+
+def _process_obj_type(lines: list[str], line: str, obj_type_match: re.Match) -> None:
+    global _line_num
+
+    name = obj_type_match.group(1)
+
+    description = ""
+    while DESCRIPTION not in line:
+        _line_num += 1
+        line = lines[_line_num].strip()
+    # The line containing "::=" always follows the description in any
+    # MIB file.
+    while "::=" not in line:
+        description += line.replace(DESCRIPTION, "").strip()
+        if description != "":
+            description += " "
+        _line_num += 1
+        line = lines[_line_num].strip()
+    # Remove multiple spaces and strip white space.
+    description = re.sub(r" +", " ", description).replace('"', "").strip()
+
+    oid_match = re.match(MIB_OID, line)
+    assert oid_match is not None
+    parent_name = oid_match.group(1)
+    parent = mib_tree[parent_name]
+    oid = oid_match.group(2)
+
+    mib_tree[name] = MibTreeElement(
+        name=name,
+        description=description,
+        oid=f"{parent.oid}.{oid}",
+        parent=parent,
+        type=MibTreeElementType.LEAF,
+    )
+
+
+# Defined this way, the "mib_tree" attribute becomes available as a "singleton"
+# for all code that imports it.
+mib_tree: dict[str, MibTreeElement] = {}
```

Comment:
uhm.. are you sure this is how you want to do this? wouldn't it be better to have a constructor class? 

This could, for instance, lead the unrecoverable errors while importing the module.

Reply:
I reworked the code so this is all in a class now.